TP3/GildedRose-Refactoring-Kata/src/test/java/com/gildedrose/GildedRoseTest.java

```diff
@@ -7,11 +7,126 @@
 class GildedRoseTest {
 
     @Test
-    void foo() {
-        Item[] items = new Item[] { new Item("foo", 0, 0) };
+    void GildedRoseTest() {
+        Item[] items = new Item[] {
+            new Item("+5 Dexterity Vest", 10, 50), //
+            new Item("item", 10, 50),
+            new Item("Aged Brie", 2, 0), //
+            new Item("Elixir of the Mongoose", 5, 7), //
+            new Item("Sulfuras, Hand of Ragnaros", 0, 80), //
+            new Item("Sulfuras, Hand of Ragnaros", -1, 80),
+            new Item("Backstage passes to a TAFKAL80ETC concert", 15, 20),
+            new Item("Backstage passes to a TAFKAL80ETC concert", 10, 49),
+            new Item("Backstage passes to a TAFKAL80ETC concert", 4, 49),
+            // this conjured item does not work properly yet
+            new Item("Conjured Mana Cake", 3, 6)
+        };
+
         GildedRose app = new GildedRose(items);
         app.updateQuality();
-        assertEquals("foo", app.items[0].name);
+
+        assertEquals("+5 Dexterity Vest, 9, 49", items[0].toString());
+        assertEquals("item, 9, 49", items[1].toString());
+        assertEquals("Aged Brie, 1, 1", items[2].toString());
+        assertEquals("Elixir of the Mongoose, 4, 6", items[3].toString());
+        assertEquals("Sulfuras, Hand of Ragnaros, 0, 80", items[4].toString());
+        assertEquals("Sulfuras, Hand of Ragnaros, 0, 80", items[5].toString());
+        assertEquals("Backstage passes to a TAFKAL80ETC concert, 14, 21", items[6].toString());
+        assertEquals("Backstage passes to a TAFKAL80ETC concert, 9, 50", items[7].toString());
+        assertEquals("Backstage passes to a TAFKAL80ETC concert, 3, 50", items[8].toString());
+        assertEquals("Conjured Mana Cake, 2, 4", items[9].toString());
+    }
+
+    @Test
+    void ItemTest1() {
+        Item item = new Item("item", 10, 50);
+        item.chooseMethod();
+        assertEquals("item, 9, 49", item.toString());
+    }
+
+    @Test
+    void ItemTest2() {
+        Item item = new Item("item", 0, 50);
+        item.chooseMethod();
+        assertEquals("item, 0, 48", item.toString());
+    }
+
+
+    @Test
+    void BackstagePassesItemTest() {
+        Item item = new Item("Backstage passes to a TAFKAL80ETC concert", 15, 20);
+        item.chooseMethod();
+        assertEquals("Backstage passes to a TAFKAL80ETC concert, 14, 21", item.toString());
+    }
+
+    @Test
+    void BackstagePassesItemTest2() {
```

Comment:
Si le but c'est de tester une certaine valeur, pourquoi utiliser chaque fois item.toString() plutôt que par exemple: 

Item[] items = new Item[] { new Item("Sulfuras, Hand of Ragnaros", 5, 50) };
        GildedRose app = new GildedRose(items);
        app.updateQuality();
        assertEquals(80, app.items[0].quality);